src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -222,47 +223,120 @@ public void close() throws IllegalStateException {
         }
     }
 
+    public static ScheduledExecutorService getExecutor() {
+        return executor;
+    }
+
+    private void basicAction(Retriable retriable, MetricArbiter.MetricMonitor metric) throws IOException {
+        checkConnection();
+        IOException exception = new IOException("Recovery attempts expired.");
+        int attemptsCount = 0;
+        int maxAttempts = configuration.getMaxRecoveryAttempts();
+
+        while (true) {
+            attemptsCount++;
+
+            try {
+                retriable.retry();
+                break;
+            } catch (Exception e) {
+                exception.addSuppressed(e);
+                LOGGER.error(e.getMessage());
+
+                if(attemptsCount >= maxAttempts) {
+                    metric.disable();
+                    throw exception;
```

Comment:
@Nikita-Smirnov-Exactpro, should we throw exception or continue trying to execute request?

Reply:
I think we should switch the liveness probe to false and continue trying.